nuxeo-distribution/nuxeo-nxr-server/src/main/resources/templates/common-base/client/scripts/upgrade_tomcat7.sh

```diff
@@ -0,0 +1,146 @@
+  if [ "$VERSIONS_FOUND" -ne 1 ]; then
+    echo -e "Cannot find Tomcat version ${2}"
+    echo
+    exit 1
+  else
+    TOMCAT_TARGET=$2
+  fi
+fi
+
+if ! WORK_FOLDER=$(mktemp -d -q /tmp/"$(basename "$0")".XXXXXX); then
+  echo "ERROR: Cannot create temp file, exiting..."
+  echo
+  exit 1
+fi
+
+TOMCAT_SOURCE=$(java -cp "${NUXEO_HOME}/lib/catalina.jar" org.apache.catalina.util.ServerInfo | grep "Server number" | sed 's/.*\(7\.0.[0-9]*\).*/\1/g')
+TOMCAT_NUXEO_DEFAULT=$(grep tomcat.version "${NUXEO_HOME}/templates/nuxeo.defaults" | cut -d "=" -f 2)
+
+echo "NUXEO_HOME is ${NUXEO_HOME}"
+echo "TEMPORARY WORK FOLDER is ${WORK_FOLDER}"
+echo "TOMCAT source version (from libs) is ${TOMCAT_SOURCE}"
+echo "TOMCAT source version (from nuxeo.defaults) is ${TOMCAT_NUXEO_DEFAULT}"
+if [ "${TOMCAT_SOURCE}" = "${TOMCAT_NUXEO_DEFAULT}" ]; then
+  echo "TOMCAT source versions match!"
+else
+  echo "ERROR: TOMCAT source versions don't match!"
+  exit 1
+fi
+echo "TOMCAT target version is ${TOMCAT_TARGET}"
+echo
+
+echo "Retrieving files..."
+rm -rf "${WORK_FOLDER}"
+mkdir -p "${WORK_FOLDER}"
+cd "${WORK_FOLDER}" || exit 1
+wget "${TOMCAT_ARCHIVE_URL}/v${TOMCAT_TARGET}/bin/apache-tomcat-${TOMCAT_TARGET}.tar.gz" || exit 1
+wget "${TOMCAT_ARCHIVE_URL}/v${TOMCAT_TARGET}/bin/apache-tomcat-${TOMCAT_TARGET}.tar.gz.md5" || exit 1
+wget "${TOMCAT_ARCHIVE_URL}/v${TOMCAT_TARGET}/bin/apache-tomcat-${TOMCAT_TARGET}.tar.gz.sha1" || exit 1
+wget "${TOMCAT_ARCHIVE_URL}/v${TOMCAT_TARGET}/bin/extras/tomcat-juli-adapters.jar" || exit 1
+wget "${TOMCAT_ARCHIVE_URL}/v${TOMCAT_TARGET}/bin/extras/tomcat-juli-adapters.jar.md5" || exit 1
+wget "${TOMCAT_ARCHIVE_URL}/v${TOMCAT_TARGET}/bin/extras/tomcat-juli-adapters.jar.sha1" || exit 1
+wget "${TOMCAT_ARCHIVE_URL}/v${TOMCAT_TARGET}/bin/extras/tomcat-juli.jar" || exit 1
+wget "${TOMCAT_ARCHIVE_URL}/v${TOMCAT_TARGET}/bin/extras/tomcat-juli.jar.md5" || exit 1
+wget "${TOMCAT_ARCHIVE_URL}/v${TOMCAT_TARGET}/bin/extras/tomcat-juli.jar.sha1" || exit 1
+
+echo "Checking archives..."
+# fix wrong md5 and sha1 file content
+sed -i 's/\*/ /g' "apache-tomcat-${TOMCAT_TARGET}.tar.gz.md5"; echo >> "apache-tomcat-${TOMCAT_TARGET}.tar.gz.md5"
+sed -i 's/\*/ /g' "apache-tomcat-${TOMCAT_TARGET}.tar.gz.sha1"; echo >> "apache-tomcat-${TOMCAT_TARGET}.tar.gz.sha1"
+sed -i 's/\*/ /g' tomcat-juli-adapters.jar.md5; echo >> tomcat-juli-adapters.jar.md5
+sed -i 's/\*/ /g' tomcat-juli-adapters.jar.sha1; echo >> tomcat-juli-adapters.jar.sha1
+sed -i 's/\*/ /g' tomcat-juli.jar.md5; echo >> tomcat-juli.jar.md5
+sed -i 's/\*/ /g' tomcat-juli.jar.sha1; echo >> tomcat-juli.jar.sha1
+# perform checks
+md5sum -c "apache-tomcat-${TOMCAT_TARGET}.tar.gz.md5" || exit 1
+shasum -c "apache-tomcat-${TOMCAT_TARGET}.tar.gz.sha1" || exit 1
+md5sum -c tomcat-juli-adapters.jar.md5 || exit 1
+shasum -c tomcat-juli-adapters.jar.sha1 || exit 1
+md5sum -c tomcat-juli.jar.md5 || exit 1
+shasum -c tomcat-juli.jar.sha1 || exit 1
```

Comment:
Note that `md5sum` and `shasum` are linux-specific and won't work on macOS. You may want to use `openssl md5` and `openssl sha1` if you want to be generic.